spec/models/solr_document_spec.rb

```diff
@@ -124,6 +124,36 @@
     end
   end
 
+  describe '#library_display' do
+    solr_info = { id: 30, item_display_struct:
+                [{ barcode: '9102385731231', callnumber: nil, current_location: nil,
+                   effective_location_id: '43c011f3-adeb-4c49-b357-06772aff11de',
+                   full_shelfkey: nil, home_location: 'BUS-ELECTRONIC', library: 'BUSINESS',
+                   permanent_location_code: 'BUS-ELECTRONIC', scheme: 'LC', type: 'ONLINE' }] }
+    physical_location_hash = solr_info.deep_dup
+    physical_location_hash[:item_display_struct].push({ barcode: 123456, home_location: 'BUS', library: 'BUSINESS',
+                                                        permanent_location_code: 'BUS' })
+    let(:solr_doc) { SolrDocument.new(solr_info) }
+    let(:physical_location_doc) { SolrDocument.new(physical_location_hash) }
+
+    it 'is has a library_display' do
+      expect(solr_doc.library_display).to eq('Business Library')
+    end
+
+    it 'has an eresource and phsyical location' do
+      expect(physical_location_doc.library_display).to be_nil
+    end
+  end
+
+  describe 'check sul ersource library_display' do
```

Comment:
And maybe you can come up with a context for this one? Is this testing the same paths as the test above?  If so, do we need it?

Reply:
It is testing one of the fixtures which is a sul eresouce. I came up with a context for this one too and added into the describe '#eresources_library_display_name' so they are all grouped together.